### PR TITLE
XWIKI-18500: Extension updater propose to upgrade Sandbox to 13.1 when 12.10.5 flavor is installed

### PR DESCRIPTION
**Pull request by slauriere**
- Add dependency on `xwiki-platform-uiextension-ui`, which is needed by the Sandbox application to get listed in the Applications panel via a `UserExtensionClass` object.

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-sandbox/pom.xml [hunk 1]
```diff
@@ -35,6 +35,15 @@
...
   </properties>
+  <dependencies>
+    <dependency>
+      <!-- Needed to display the Sandbox in the Applications panel list -->
+      <groupId>org.xwiki.platform</groupId>
+      <artifactId>xwiki-platform-uiextension-ui</artifactId>
+      <version>${project.version}</version>
+      <type>xar</type>
+    </dependency>
+  </dependencies>
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-sandbox/pom.xml:46]: ``xwiki-platform-uiextension-ui`` only has a sheet which I doubt is used by the Sandbox application. The class is generated by ``UIExtensionClassDocumentInitializer`` from the API module.

Comment (slauriere) [reply xwiki-platform-core/xwiki-platform-sandbox/pom.xml:46]: Wrong pick by me indeed, sorry, fixed in latest commit.

File: xwiki-platform-core/xwiki-platform-sandbox/pom.xml [hunk 2]
```diff
@@ -35,6 +35,15 @@
     <!-- Name to display by the Extension Manager -->
     <xwiki.extension.name>Sandbox Application</xwiki.extension.name>
   </properties>
+  <dependencies>
+    <dependency>
+      <!-- Needed to display the Sandbox in the Applications panel list -->
+      <groupId>org.xwiki.platform</groupId>
+      <artifactId>xwiki-platform-uiextension-api</artifactId>
+      <version>${project.version}</version>
+      <type>xar</type>
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-sandbox/pom.xml:44]:
```suggestion
```

Wrong type.

**Review by mflorea — CHANGES_REQUESTED** [review]

File: xwiki-platform-core/xwiki-platform-sandbox/pom.xml (hunk 2 above)
Comment (slauriere) [reply xwiki-platform-core/xwiki-platform-sandbox/pom.xml:44]: Damn, sorry and thank you

**Review by mflorea — APPROVED** [review]

**Pull request merged**